lib/records/fhir.rb

```diff
@@ -299,9 +302,12 @@ def self.medications(prescription, fhir_record, patient, encounter)
           'patient' => {'reference'=> "Patient/#{patient.fullUrl}"},
           'encounter' => {'reference'=> "Encounter/#{encounter.fullUrl}"},
           'dateWritten' => convertFhirDateTime(prescription['time']),
-          'reasonReference' => {'reference'=> "Condition/#{reason.fullUrl}"},
+          'reasonReference' => [] ,
           'activity' => []
         })
+        reasons.each do |r|
+          medOrder.reasonReference << {'reference'=> "Condition/#{r.fullUrl}"}
```

Comment:
You can only use the hashes inside the `FHIR::MedicationOrder.new`... if you add a field subsequently, you need to create the object. In this case, something like `FHIR::Reference.new`